### PR TITLE
add Warning type to message,add function for determine valid date

### PR DESCRIPTION
**Pull request by okrdima**

### CODE REVIEW

File: src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js [hunk 1]
```diff
@@ -193,7 +206,7 @@ const ProductAdvancedView = (props) => {
     } catch (error) {
       messageDispatch({
         type: types.OPEN_ERROR_MESSAGE,
-        payload: error
+        payload: 'Product can not move to cart'
```
Comment (GingerBoy14) [inline src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js:209]: Write or can't or can not (same style everywhere)

**Review by GingerBoy14 — CHANGES_REQUESTED** [review]

File: src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js [hunk 2]
```diff
@@ -149,12 +148,21 @@ const ProductAdvancedView = (props) => {
...
       const category = await getProductCategory()
-      await setData(COLLECTIONS.CATEGORIES, category.docs[0].id, {
-        spent: category.docs[0].data().spent + parseInt(data.price)
-      })
+      if (category.docs.length === 0) {
+        await setData(COLLECTIONS.PURCHASES, id, {
+          ...product,
+          category: 'Other'
+        })
+        categoryIs = true
```
Comment (GingerBoy14) [inline src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js:159]: What the purpose of using `categoryIs` var, but don't to enter messageDispatch right here

**Review by GingerBoy14 — CHANGES_REQUESTED** [review]

File: src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js [hunk 3]
```diff
@@ -149,12 +148,21 @@ const ProductAdvancedView = (props) => {
       await setData(COLLECTIONS.WALLETS, wallet.id, {
         balance: wallet.balance - product.price
       })
-      /*
-        set spended money to category spendings*/
+      /*set spended money to category spendings*/
+      let categoryIs = false
```
Comment (5P4C3GR33D) [inline src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js:152]: Wrong naming.
Not 'categoryIs', but 'isCategory'. C:

Comment (5P4C3GR33D) [reply src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js:152]: Or even `hasCategory`.

**Review by olenashevchuk — APPROVED** [review]

**Pull request merged**